processor/filterprocessor/factory.go

```diff
@@ -51,6 +53,13 @@ func createMetricsProcessor(
 	cfg configmodels.Processor,
 	nextConsumer consumer.MetricsConsumer,
 ) (component.MetricsProcessor, error) {
-	oCfg := cfg.(*Config)
-	return newFilterMetricProcessor(nextConsumer, oCfg)
+	fp, err := newFilterMetricProcessor(cfg.(*Config))
+	if err != nil {
+		return nil, err
+	}
+	return processorhelper.NewMetricsProcessor(
+		cfg,
+		nextConsumer,
+		fp.filterMetrics,
```

Comment:
It may just be my background, so feel free to ignore this comment, but I'm never a big fan of passing instance functions as arguments. The cognitive reasoning of how this works becomes much more difficult to me for some reason.

One possible alternative would be to create a simple [one-method interface](https://golang.org/doc/effective_go.html#interface-names) `MetricsConsumerNew` (joking about that name) that has just a `ConsumeMetrics` function and no other required boilerplate.

(even if this isa good idea, the overhead of migrating to a new interface is likely not worth it anyway)

Reply:
Tried this. Seems reasonable.